rust/core-lib/src/linewrap.rs

```diff
@@ -82,6 +82,21 @@ pub(crate) struct InvalLines {
     pub(crate) new_count: usize,
 }
 
+/// Detailed information about changes to linebreaks, used to generate
+/// invalidation information. The logic behind invalidation is different
+/// depending on whether we're updating breaks after an edit, or continuing
+/// a bulk rewrapping task. This is shared between the two cases, and contains
+/// the information relevant to both of them.
+struct WrapSummary {
+    start_line: usize,
+    /// Total number of invalidated lines; this is meaningless in the after_edit case.
+    inval_count: usize,
+    /// The total number of new (hard + soft) breaks in the wrapped region.
+    new_count: usize,
+    /// The number of new soft breaks.
+    new_soft: usize,
+}
```

Comment:
This looks like a C-ism. Is any of the other fields unique to one case? If so, I'd probably turn this into a struct enum just to err on the safe side. If not, maybe consider making `invalid_count` a `Option<usize>`?

Reply:
The problem is  the function we're calling doesn't know which of the two callers it's dealing with. I've taken the approach of just making sure I point out the tricky bits and hopefully explain the situation. Doesn't feel great or anything but felt like the best option. 😞

Reply:
Hmm yeah, reading the code it makes a bit more sense to have it be this way.